### PR TITLE
Optional schema validation

### PR DESCRIPTION
**Pull request by kittaakos**
#### What it does

[Made the preference schema validation optional.](https://github.com/eclipse-theia/theia/commit/361f1e10f84b7061e0331d6b4147f13863cf589a) 
 - It costs time at startup, and it only logs warnings.
 - Can be disabled with the `shouldValidateSchema` preference config.

This PR contains a commit with an example. I can drop the example commit from the PR.

#### How to test

Start the app, and you will see these in the log:
```
2022-05-21T16:37:49.331Z root INFO Preference schema validation is disabled.
2022-05-21T16:37:49.331Z root INFO Preference schema validation is disabled.
2022-05-21T16:37:49.331Z root INFO Preference schema validation is disabled.
2022-05-21T16:37:49.332Z root INFO Preference schema validation is disabled.
2022-05-21T16:37:49.332Z root INFO [b102b270-04b0-4af7-93f7-d8b1f0e2c9ef][vscode.css-language-features]: Loaded contributions.
2022-05-21T16:37:49.332Z root INFO [b102b270-04b0-4af7-93f7-d8b1f0e2c9ef][vscode.debug-auto-launch]: Loaded contributions.
2022-05-21T16:37:49.333Z root INFO [b102b270-04b0-4af7-93f7-d8b1f0e2c9ef][vscode.debug-server-ready]: Loaded contributions.
2022-05-21T16:37:49.333Z root INFO Preference schema validation is disabled.
2022-05-21T16:37:49.334Z root INFO [b102b270-04b0-4af7-93f7-d8b1f0e2c9ef][vscode.docker]: Loaded contributions.
2022-05-21T16:37:49.334Z root INFO Preference schema validation is disabled.
```

Please review. Thank you! 🌻 

#### Review checklist

- [ ] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by msujew** [comment]
@kittaakos I'm alright with having an option to disable schema validation, but having it behind an overridable method seems overengineered for me. What do you think about a frontend config option instead?

**Comment by kittaakos** [comment]
> I'm alright with having an option to disable schema validation

Thank you!

> What do you think about a frontend config option instead?

Sure. Do you mean `FrontendApplicationConfig`?

**Comment by msujew** [comment]
> Do you mean FrontendApplicationConfig?

Yes exactly. It allows downstream users to easier disable those validations, since the json can be easily modified during the build process. Having something like this locked behind code basically only allows manual manipulation, which is then app-wide.

File: packages/core/src/browser/preferences/preference-contribution.ts
Comment (colin-grant-work) [inline packages/core/src/browser/preferences/preference-contribution.ts:191]:
```suggestion
    protected validateSchema(schema: PreferenceSchema): void {
```

Comment (kittaakos) [reply packages/core/src/browser/preferences/preference-contribution.ts:191]: I would keep it private. This PR does not want to allow custom preference schema validation.

Comment (colin-grant-work) [reply packages/core/src/browser/preferences/preference-contribution.ts:191]: > I would keep it private. This PR does not want to allow custom preference schema validation.

Do you have a compelling reason why the framework should _not_ allow custom preference schema validation, if for example a downstream project prefers a different library to AJV or wants to implement a custom check?

**Comment by kittaakos** [comment]
Please review.

**Review by msujew — APPROVED** [review]
The changes look good to me! Don't forget to revert your test commit before merging 👍

**Comment by kittaakos** [comment]
> Don't forget to revert your test commit before merging 👍

Do you want me to drop the examples customization?

**Comment by msujew** [comment]
@kittaakos Yes, more specifically 43456de, after all we want to have validations for our example app (for us to able to test that the validation works as expected)

**Comment by colin-grant-work** [comment]
@kittaakos, it looks like you still have committer's rights, so you're welcome to do merges yourself, once a PR is approved.

**Comment by kittaakos** [comment]
> it looks like you still have committer's rights

No, I do not have it 😊 The Eclipse page is misleading. But anyway, thanks for checking it and chiming in for the PR.

**Pull request merged**
